### PR TITLE
statistics: fix wrong NDV in the global stats when to disable async-merge-global-stats

### LINKED ISSUE #53752: wrong NDV in the global stats when to disable async-merge-global-stats
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
set @@tidb_enable_async_merge_global_stats=OFF;
CREATE TABLE employees  (
    id INT NOT NULL AUTO_INCREMENT PRIMARY KEY,
    fname VARCHAR(25) NOT NULL,
    lname VARCHAR(25) NOT NULL,
    store_id INT NOT NULL,
    department_id INT NOT NULL
)

PARTITION BY RANGE(id)  (
    PARTITION p0 VALUES LESS THAN (5),
    PARTITION p1 VALUES LESS THAN (10),
    PARTITION p2 VALUES LESS THAN (15),
    PARTITION p3 VALUES LESS THAN MAXVALUE
);



INSERT INTO employees(FNAME,LNAME,STORE_ID,DEPARTMENT_ID) VALUES
    ('Bob', 'Taylor', 3, 2), ('Frank', 'Williams', 1, 2),
    ('Ellen', 'Johnson', 3, 4), ('Jim', 'Smith', 2, 4),
    ('Mary', 'Jones', 1, 1), ('Linda', 'Black', 2, 3),
    ('Ed', 'Jones', 2, 1), ('June', 'Wilson', 3, 1),
    ('Andy', 'Smith', 1, 3), ('Lou', 'Waters', 2, 4),
    ('Jill', 'Stone', 1, 4), ('Roger', 'White', 3, 2),
    ('Howard', 'Andrews', 1, 2), ('Fred', 'Goldberg', 3, 3),
    ('Barbara', 'Brown', 2, 3), ('Alice', 'Rogers', 2, 2),
    ('Mark', 'Morgan', 3, 3), ('Karen', 'Cole', 3, 2);

ANALYZE TABLE employees;

select * from employees;

SHOW STATS_HISTOGRAMS WHERE TABLE_NAME='employees';

alter table employees truncate   partition p0 ; 

select * from employees;

SHOW STATS_HISTOGRAMS WHERE TABLE_NAME='employees';



analyze table employees partition p3;

select * from employees;

SHOW STATS_HISTOGRAMS WHERE TABLE_NAME='employees'

```

### 2. What did you expect to see? (Required)

```
+---------+------------+----------------+---------------+----------+---------------------+----------------+------------+--------------+-------------+-------------+-----------------+----------------+----------------+---------------+
| Db_name | Table_name | Partition_name | Column_name   | Is_index | Update_time         | Distinct_count | Null_count | Avg_col_size | Correlation | Load_status | Total_mem_usage | Hist_mem_usage | Topn_mem_usage | Cms_mem_usage |
+---------+------------+----------------+---------------+----------+---------------------+----------------+------------+--------------+-------------+-------------+-----------------+----------------+----------------+---------------+
| test    | employees  | global         | id            | 0        | 2024-06-03 14:36:02 | 14             | 0          | 8.0          | 0.0         | allEvicted  | 0               | 0              | 0              | 0             |
| test    | employees  | global         | fname         | 0        | 2024-06-03 14:36:02 | 14             | 0          | 5.29         | 0.0         | allEvicted  | 0               | 0              | 0              | 0             |
| test    | employees  | global         | lname         | 0        | 2024-06-03 14:36:02 | 13             | 0          | 6.63         | 0.0         | allEvicted  | 0               | 0              | 0              | 0             |
| test    | employees  | global         | store_id      | 0        | 2024-06-03 14:36:02 | 3              | 0          | 5.08         | 0.0         | allEvicted  | 0               | 0              | 0              | 0             |
| test    | employees  | global         | department_id | 0        | 2024-06-03 14:36:02 | 4              | 0          | 5.08         | 0.0         | allEvicted  | 0               | 0              | 0              | 0             |
+---------+------------+----------------+---------------+----------+---------------------+----------------+------------+--------------+-------------+-------------+-----------------+----------------+----------------+---------------+
```
### 3. What did you see instead (Required)

```
+---------+------------+----------------+---------------+----------+---------------------+----------------+------------+--------------+-------------+-------------+-----------------+----------------+----------------+---------------+
| Db_name | Table_name | Partition_name | Column_name   | Is_index | Update_time         | Distinct_count | Null_count | Avg_col_size | Correlation | Load_status | Total_mem_usage | Hist_mem_usage | Topn_mem_usage | Cms_mem_usage |
+---------+------------+----------------+---------------+----------+---------------------+----------------+------------+--------------+-------------+-------------+-----------------+----------------+----------------+---------------+
| test    | employees  | global         | id            |        0 | 2024-06-03 15:04:21 |              0 |          0 |            8 |           0 | allEvicted  |               0 |              0 |              0 |             0 |
| test    | employees  | global         | fname         |        0 | 2024-06-03 15:04:21 |              0 |          0 |         5.29 |           0 | allEvicted  |               0 |              0 |              0 |             0 |
| test    | employees  | global         | lname         |        0 | 2024-06-03 15:04:21 |              0 |          0 |         6.63 |           0 | allEvicted  |               0 |              0 |              0 |             0 |
| test    | employees  | global         | store_id      |        0 | 2024-06-03 15:04:21 |              0 |          0 |         5.08 |           0 | allEvicted  |               0 |              0 |              0 |             0 |
| test    | employees  | global         | department_id |        0 | 2024-06-03 15:04:21 |              0 |          0 |         5.08 |           0 | allEvicted  |               0 |              0 |              0 |             0 |
+---------+------------+----------------+---------------+----------+---------------------+----------------+------------+--------------+-------------+-------------+-----------------+----------------+----------------+---------------+
5 rows in set (0.00 sec)
```

### 4. What is your TiDB version? (Required)

master

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #53752

Problem Summary:

### What changed and how does it work?

blockMergeGlobalStats doesn't notice that FMSketch contains empty objectives in all partition's stats. so we need to find nil to init global rightly.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

run as the PR's description.

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/statistics/handle/globalstats/global_stats_test.go [hunk 1]
```diff
@@ -959,3 +959,75 @@ func TestGlobalStatsAndSQLBindingWithConcurrency(t *testing.T) {
 	tk.MustExec("set global tidb_merge_partition_stats_concurrency=2")
 	testGlobalStatsAndSQLBinding(tk)
 }
+
+func TestBlockCheckFMSketch(t *testing.T) {
```
Comment (hi-rustin) [inline pkg/statistics/handle/globalstats/global_stats_test.go:963]: It seems that even if I revert your change I can still pass the tests. So maybe the test case doesn't cover the change?

Comment (hi-rustin) [inline pkg/statistics/handle/globalstats/global_stats_test.go:963]:
```suggestion
func TestBlockMergeFMSketch(t *testing.T) {
```

File: pkg/statistics/handle/globalstats/global_stats_test.go [hunk 2]
```diff
@@ -959,3 +959,75 @@ func TestGlobalStatsAndSQLBindingWithConcurrency(t *testing.T) {
...
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	tk.MustExec("set @@tidb_enable_async_merge_global_stats=OFF;")
+	defer func() {
+		tk.MustExec("set @@tidb_enable_async_merge_global_stats=ON;")
+	}()
+	checkFMSketch(tk)
+}
+
+func TestAsyncCheckFMSketch(t *testing.T) {
```
Comment (hi-rustin) [inline pkg/statistics/handle/globalstats/global_stats_test.go:974]:
```suggestion
func TestAsyncMergeFMSketch(t *testing.T) {
```

File: pkg/statistics/handle/globalstats/global_stats_test.go [hunk 3]
```diff
@@ -959,3 +959,75 @@ func TestGlobalStatsAndSQLBindingWithConcurrency(t *testing.T) {
...
+    ('Mary', 'Jones', 1, 1), ('Linda', 'Black', 2, 3),
+    ('Ed', 'Jones', 2, 1), ('June', 'Wilson', 3, 1),
+    ('Andy', 'Smith', 1, 3), ('Lou', 'Waters', 2, 4),
+    ('Jill', 'Stone', 1, 4), ('Roger', 'White', 3, 2),
+    ('Howard', 'Andrews', 1, 2), ('Fred', 'Goldberg', 3, 3),
+    ('Barbara', 'Brown', 2, 3), ('Alice', 'Rogers', 2, 2),
+    ('Mark', 'Morgan', 3, 3), ('Karen', 'Cole', 3, 2);`)
+	tk.MustExec("ANALYZE TABLE employees;")
+	tk.MustExec("select * from employees;")
+	tk.MustExec("alter table employees truncate   partition p0 ; ")
```
Comment (hi-rustin) [inline pkg/statistics/handle/globalstats/global_stats_test.go:1002]:
```suggestion
	tk.MustExec("alter table employees truncate partition p0;")
```

File: pkg/statistics/handle/globalstats/global_stats_test.go (hunk 1 above)
Comment (hawkingrei) [reply pkg/statistics/handle/globalstats/global_stats_test.go]: You should disable the ```tidb_enable_async_merge_global_stats```.

Comment (hi-rustin) [reply pkg/statistics/handle/globalstats/global_stats_test.go]: Yes, I have. I've run the TestBlockCheckFMSketch. 
1. I revert your change. 
3. I ran the TestBlockCheckFMSketch, and it still passed.

Comment (hawkingrei) [reply pkg/statistics/handle/globalstats/global_stats_test.go]: I find this problem, only it is reproduced in the real tikv.

**Review by winoros — APPROVED** [review]

**Review by hi-rustin — APPROVED** [review]

**Pull request merged**